### PR TITLE
feat(KONFKUX-4010): Add integration service pipeline to run image-controller e2e tests in Konflux CI

### PR DESCRIPTION
**Pull request by flacatus**
QE docs for integration tests

https://konflux-ci-test.gitbook.io/konflux-ci-test-system/onboarding-integration-tests/add_integration_tests#requirements

### CODE REVIEW

**Comment by flacatus** [comment]
we need first to merge: https://github.com/konflux-ci/e2e-tests/pull/1411

**Review by tnevrlka — COMMENTED** [review]
Thank you for the contribution! Please change the commit message so the lines are 72 chars max. Thank you!

File: integration-tests/pipelines/konflux-e2e-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+kind: Pipeline
+metadata:
+  name: konflux-e2e-tests
+spec:
+  description: |-
+    This pipeline automates the process of running end-to-end tests for Konflux
+    using a ROSA (Red Hat OpenShift Service on AWS) cluster. The pipeline provisions
+    the ROSA cluster, installs Konflux using the infra-deployments, runs the image-controller tests, collects artifacts,
+    and finally deprovisions the ROSA cluster.
+  params:
```
Comment (mmorhun) [inline integration-tests/pipelines/konflux-e2e-tests.yaml:12]: How can I change some params for the pipeline?

File: integration-tests/pipelines/konflux-e2e-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    - name: SNAPSHOT
+      description: 'The JSON string representing the snapshot of the application under test.'
+      default: '{"components": [{"name":"test-app", "containerImage": "quay.io/example/repo:latest"}]}'
+      type: string
+    - name: test-name
+      description: 'The name of the test corresponding to a defined Konflux integration test.'
+      default: ''
+    - name: ocp-version
```
Comment (mmorhun) [inline integration-tests/pipelines/konflux-e2e-tests.yaml:20]: Is it bumped automatically?

Comment (flacatus) [reply integration-tests/pipelines/konflux-e2e-tests.yaml:20]: unfortunetly no. We need to stick to konflux version

File: integration-tests/pipelines/konflux-e2e-tests.yaml (hunk 1 above)
Comment (flacatus) [reply integration-tests/pipelines/konflux-e2e-tests.yaml:12]: we need to change it via integration tests

**Review by mmorhun — APPROVED** [review]

File: integration-tests/pipelines/konflux-e2e-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+          - name: revision
+            value: main
+          - name: pathInRepo
+            value: common/tasks/test-metadata/0.1/test-metadata.yaml
+      params:
+        - name: SNAPSHOT
+          value: $(params.SNAPSHOT)
+        - name: test-name
+          value: $(context.pipelineRun.name)
+    - name: create-oci-container
```
Comment (tisutisu) [inline integration-tests/pipelines/konflux-e2e-tests.yaml:71]: Will it be better to run this task after `provision-rosa` task, if `provision-rosa` task fails, it is of no use right?

Comment (flacatus) [reply integration-tests/pipelines/konflux-e2e-tests.yaml:71]: @tisutisu there is a plan to collect rosa task logs in the oci container

Comment (tisutisu) [reply integration-tests/pipelines/konflux-e2e-tests.yaml:71]: It makes sense to keep it then.

**Comment by flacatus** [comment]
@psturc @tisutisu can you help to understand this failure?

**Comment by psturc** [comment]
> @psturc @tisutisu can you help to understand this failure?

looks like sprayproxy issue I mentioned [here](https://redhat-internal.slack.com/archives/C02FANRBZQD/p1727791884774909?thread_ts=1727777584.068689&cid=C02FANRBZQD)

but when I look at [other recent CI jobs](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/redhat-appstudio_infra-deployments/4624/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1841154205176303616), the issue is no longer there.

IDK maybe bad timing, let's try again
/retest

**Pull request merged**